object-store-client-play-26/src/main/scala/uk/gov/hmrc/objectstore/client/ObjectStoreClient.scala

```diff
@@ -76,6 +77,13 @@ class ObjectStoreClient[F[_], REQ_BODY, RES, RES_BODY](
     F.flatMap(client.get(location, headers()))(read.toObjectListing)
   }
 
+  private val hcConfig = HeaderCarrier.Config.fromConfig(ConfigFactory.load())
+
   private def headers(additionalHeaders: (String, String)*)(implicit hc: HeaderCarrier): List[(String, String)] =
-    hc.copy(authorization = Some(Authorization(config.authorizationToken))).headers.toList ++ additionalHeaders
+    hc
+      .copy(authorization = Some(Authorization(config.authorizationToken)))
+      .headersForUrl(hcConfig)(url)
```

Comment:
What's the intent of this line?

Reply:
`headers` is no longer available, you have to use `headersForUrl` and provide the target url. This is how it knows which headers to forward - e.g. Authorization headers are only forwarded to internal urls now.
Config is required (to define the internal hosts) - which is available on classpath from httpverbs reference.conf.

Reply:
Isn't this quite fragile? A client can provide a known URL pattern (from reference.conf) and obtain all the headers and subsequently forward the headers wherever the client wants.

Reply:
It's up to a client if it wants to send all headers. We're just reducing the risk of doing it by accident. A client can also call `headers(Authorization, ...)` to get the headers to forward on - that's the expected pattern since it's explicit.